src/features/dataSourceSelector/lib/hooks/useRefetchSuggestions.ts

```diff
@@ -1,13 +1,24 @@
-import {PipelineId} from '@/entities/pipeline/model';
-import {useDispatch} from '@/infra/providers/redux';
 import {useEffect} from 'react';
+import {useSelector} from 'react-redux';
+
+import {
+  PipelineId,
+  selectDefaultWidgetDataShape,
+  selectMappers,
+} from '@/entities/pipeline/model';
+import {useDispatch} from '@/infra/providers/redux';
 import {
   fetchPipelineDataSourceSuggestions
 } from '@/entities/dataSourceSuggestion/model/fetchPipelineDataSourceSuggestions';
 
 export const useRefetchSuggestions = ({pipelineId}: {pipelineId: PipelineId}) => {
   const dispatch = useDispatch();
+  const dataShape = useSelector(state => selectDefaultWidgetDataShape(state, pipelineId));
+  const mappers = useSelector(state => selectMappers(state, pipelineId));
   useEffect(() => {
-    dispatch(fetchPipelineDataSourceSuggestions({pipelineId}));
-  }, [dispatch, pipelineId]);
+    dispatch(fetchPipelineDataSourceSuggestions({
+      pipelineId,
+      entity: mappers![dataShape!].config.entity,
```

Comment:
here we assert that every mapper will have "entity" as config field

would be way better to assert that we have just one entity (seeing that our current UI MVP is built on that assumption right now)

Reply:
right now we do have two mappers that have "entity" field in their config, but at some point we are bound to have less generic mappers (e.g. ones that only accept one specific entity)

Reply:
We had a call. In MVP we had just one entity available in mapper, but later we can define several. All of them should be choosen, so all entity type datasources should be returned